### DISCUSSION TITLE
Renaming Selector to Fallback

### DISCUSSION THREAD

**Discussion by lostptr** [bitbrain/beehave#79] [Ideas]
Since we are planning on renaming some nodes and breaking compatibilities, I would like to make a suggestion.

The `Selector` node, in my opinion, does not have a very descriptive name. It "selects" what node to execute. Doesn't all `Composite` nodes do that?

My suggestion is to rename it to `Fallback`, since it describes what it does: try that action, if it does not work try this action and so on.

What do you think? Do you have another name? Is it a necessary change?

**Comment by bitbrain** [comment]
The node names are quite common and I'd prefer to keep them consistent with other engines, see Unreal Engine: https://docs.unrealengine.com/4.27/en-US/InteractiveExperiences/ArtificialIntelligence/BehaviorTrees/BehaviorTreeNodeReference/BehaviorTreeNodeReferenceComposites/

Also, "Fallback" is usually used in the context of "default to this value".

**Reply by lostptr** [reply]
Yeah, I've only ever used this BehaviorTree.CPP library so maybe it's just a matter of me getting used to it 😆
